todo/src/components/TodoForm.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { addTodo, deleteCompleted } from '../actions/actions';
+
+class TodoForm extends React.Component {
+  state = { todo: '' };
+
+  changeHandler = e => {
+    e.preventDefault();
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  submitTodo = e => {
+    e.preventDefault();
+    this.props.addTodo(this.state.todo);
+    this.setState({
+      todo: ''
+    });
+  };
+
+  render() {
+    return (
+      <form onSubmit={e => this.submitTodo(e)}>
+        <input
+          type='text'
+          value={this.state.todo}
+          name='todo'
+          onChange={this.changeHandler}
+          placeholder='new todo'
+          required
+        />
+        <div className='buttons'>
+          <button className='add-todo' type='submit'>
+            Add Todo
+          </button>
+          <button
+            className='delete-completed'
+            type='button'
+            onClick={this.props.deleteCompleted}
+          >
+            Delete Completed
+          </button>
+        </div>
+      </form>
+    );
+  }
+}
+
+const mapStateToProps = state => {
+  return {
+    todos: state.todos
+  };
+};
+
+export default connect(
+  mapStateToProps,
```

Comment:
this is where it's interesting -- unless you actually need something from state within the specific component (you don't with this component), you can not define a MSTP function at all and simply pass in `null` here since the connect function still expects an argument in the MSTP place